### ISSUE TITLE
Missing key tmp_var when loading 

### ISSUE THREAD

**Issue by Binbose** [VLL-HD/FrEIA#10]
I used the MNIST example to train 10 model, each with the same architecture and different initializations.

I then loop over these model, want to load and benchmark each one like so:

```
model = inn_model.MNIST_cINN(0)
model.cuda()
with torch.no_grad():
    for model_name in model_list_names:
        
        state_dict = {k:v for k,v in torch.load('inn_model/{}'.format(model_name)).items() if 'tmp_var' not in k}
        
        model.load_state_dict(state_dict)
        
        SAMPLING MODEL AND BENCHMARKING
```

However, I get this error in the model.load_state_dict(state_dict) line:

```
RuntimeError: Error(s) in loading state_dict for MNIST_cINN:
	Unexpected key(s) in state_dict: "cinn.tmp_var_0", "cinn.tmp_var_1", "cinn.tmp_var_2", "cinn.tmp_var_3", "cinn.tmp_var_4", "cinn.tmp_var_5", "cinn.tmp_var_6", "cinn.tmp_var_7", "cinn.tmp_var_8", "cinn.tmp_var_9", "cinn.tmp_var_10", "cinn.tmp_var_11", "cinn.tmp_var_12", "cinn.tmp_var_13", "cinn.tmp_var_14", "cinn.tmp_var_15", "cinn.tmp_var_16", "cinn.tmp_var_17", "cinn.tmp_var_18", "cinn.tmp_var_19", "cinn.tmp_var_20", "cinn.tmp_var_21", "cinn.tmp_var_22", "cinn.tmp_var_23", "cinn.tmp_var_24", "cinn.tmp_var_25", "cinn.tmp_var_26", "cinn.tmp_var_27", "cinn.tmp_var_29", "cinn.tmp_var_30", "cinn.tmp_var_31", "cinn.tmp_var_32", "cinn.tmp_var_33", "cinn.tmp_var_34", "cinn.tmp_var_35", "cinn.tmp_var_36", "cinn.tmp_var_37", "cinn.tmp_var_38", "cinn.tmp_var_40", "cinn.tmp_var_41", "cinn.tmp_var_42", "cinn.tmp_var_43", "cinn.tmp_var_44". 
```

If instead, I just create a new model inside the loop, it works. However, then sampling the model gives me an CUDA out of memory error after benchmarking some model. Probably, because somewhere are still references to the generated data which do not get deleted.

**Comment by ardizzone** [comment]
Hi,

thank you for bringing this up! This is a known problem, and we will fix it permanently in the next major update.

For now, I can offer two workarounds:

 ---
Option 1: Load the state dict with `strict=False`
```python
model.load_state_dict(state_dict, strict=False)
```
But be aware, if other parameters are also missing e.g. because the architecture changed, there will be no error.

 ---
Option 2: make a state dict without those variables
```python
filtered_dict = {k:v for k,v in state_dict.items() if 'tmp_var' not in k}
```
then load that dict instead. You can also `del` the old state dict, to free up the buffer variables.
This should be the preferred solution I think, but both work equally.

---
I will leave the issue open until we fix it definitively.

**Comment by fdraxler** [comment]
This should be closed with [afd7adb9](https://github.com/ardizzone/FrEIA/commit/afd7adb9045c180946f63bd0076193708d3b586e).
